.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+# Trigger the workflow on push or pull request, but only for the master branch
+on:
+  pull_request:
+  push:
+    branches: ["master"]
+
+jobs:
+  generateMatrix:
+    name: "Generate matrix from cabal"
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout base repo
+        uses: actions/checkout@v3
```

Comment:
Dependabot will probably scream soon that `v4` is available.

Reply:
No it will just open a very helpful PR that will get merged promptly. :)